### PR TITLE
Simplify sdk

### PR DESCRIPTION
**Pull request by haiodo**
## Description

This PR simplifies the SDK, by using the context.Context to pass
data between Endpoints in the chain.

By doing this, and a few other minor changes, the Endpoints used
in CompositeEndpoint just need to be normal implementers of

networkservice.NetworkServiceServer

Among other things, this means that CompositeEndpoints can be nested
within each other.

Note: Updates to sdk/README.md will be forthcoming.

Work is related to #1324 #1508 

## How Has This Been Tested?

- [x] Covered by existing integration testing
- [ ] Added integration testing to cover
- [x] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: controlplane/pkg/apis/local/connection/mechanism_helpers.go [hunk 1]
```diff
@@ -16,7 +17,12 @@ import (
 func NewMechanism(t MechanismType, name, description string) (*Mechanism, error) {
 	inodeNum, err := tools.GetCurrentNS()
 	if err != nil {
-		return nil, err
+		if runtime.GOOS == "darwin" {
```
[the remark this answers is not in the document]
Comment (haiodo) [reply controlplane/pkg/apis/local/connection/mechanism_helpers.go]: Removed

File: controlplane/pkg/sidecars/nsm-monitor.go [hunk 2]
```diff
@@ -105,13 +109,14 @@ func (c *nsmMonitorApp) SetHandler(listener NSMMonitorHandler) {
...
+	var tracer opentracing.Tracer
 	if c.helper == nil || c.helper.IsEnableJaeger() {
-		tracer, closer := tools.InitJaeger("nsm-monitor")
+		tracer, tracingCloser = tools.InitJaeger("nsm-monitor")
 		opentracing.SetGlobalTracer(tracer)
-		defer func() { _ = closer.Close() }()
 	}
 
-	go c.beginMonitoring()
+	go c.beginMonitoring(tracingCloser)
```
[the remark this answers is not in the document]
Comment (haiodo) [reply controlplane/pkg/sidecars/nsm-monitor.go]: Done

File: controlplane/pkg/sidecars/nsm-monitor.go [hunk 3]
```diff
@@ -140,18 +148,15 @@ func (c *nsmMonitorApp) beginMonitoring() {
 
 		monitorClient, err := local.NewMonitorClient(nsmClient.NsmConnection.GrpcClient)
 		if err != nil {
-			logrus.Fatalf(nsmMonitorLogWithParamFormat, "failed to start monitor client", err)
+			logrus.Errorf(nsmMonitorLogWithParamFormat, "failed to start monitor client", err)
```
[the remark this answers is not in the document]
Comment (haiodo) [reply controlplane/pkg/sidecars/nsm-monitor.go]: Done.

File: controlplane/pkg/sidecars/nsm-monitor.go [hunk 4]
```diff
@@ -140,18 +148,15 @@ func (c *nsmMonitorApp) beginMonitoring() {
...
 
 			c.waitRetry()
 			continue
```
[the remark this answers is not in the document]
Comment (haiodo) [reply controlplane/pkg/sidecars/nsm-monitor.go]: Thanks.

File: go.mod [hunk 5]
```diff
@@ -2,6 +2,10 @@ module github.com/networkservicemesh/networkservicemesh
 
 // Use ./scripts/switch_k8s_version.sh to change k8s version
 
+replace gonum.org/v1/gonum => github.com/gonum/gonum v0.0.0-20190331200053-3d26580ed485
```
[the remark this answers is not in the document]
Comment (haiodo) [reply go.mod]: Done. Thanks.

**Comment by edwarnicke** [comment]
@haiodo This looks good to go to me once it passes CI.

**Comment by haiodo** [comment]
@edwarnicke @nickolaev @lobkovilya It seems all issues related to SDK are fixed, Please take a look. Thanks.

**Comment by haiodo** [comment]
rebased on master

**Pull request merged**